sapio-base/src/effects/mod.rs

```diff
@@ -25,33 +24,43 @@ pub type EffectPath = ReversePath<PathFragment>;
 #[derive(Debug)]
 pub enum EffectDBError {
     /// Error was from Deserialization
-    SerializationError(serde_json::Error),
+    SerializationError(sapio_data_repr::Error),
 }
 
-impl From<serde_json::Error> for EffectDBError {
-    fn from(e: serde_json::Error) -> Self {
+impl From<sapio_data_repr::Error> for EffectDBError {
+    fn from(e: sapio_data_repr::Error) -> Self {
         EffectDBError::SerializationError(e)
     }
 }
 /// A Generic Trait for EffectDB Functionality
 pub trait EffectDB {
-    /// internal implementation to retrieve a JSON for the path
-    fn get_value<'a>(
```

Comment:
i think the name "get_value" still makes sense since it's get_value as in key/value, not as in json_value?